### PR TITLE
Settings hackathon

### PR DESCRIPTION
**Pull request by zineanteoh**
## Summary 

1. Added vscode intellisence feature for css via typescript-plugin-css-modules 
(so CMD+click on `styles` now redirects us to the css file, rather than .d.ts)

2. Refactored a bunch of code related to non-existing `styles`

3. Created `<HackathonSettings />` as a place for organizers to view/edit these important dates: 
- Hackathon Start
- Hackathon End
- Judging Start
- Judging End

4. Created a new `hackathon` collection that stores a single document: 
```json
{
  "HACKATHON_START": "10/28/2023 05:00 PM",
  "HACKATHON_END": "10/29/2023 12:00 PM",
  "JUDGING_START": "10/29/2023 01:30 PM",
  "JUDGING_END": "10/29/2023 04:30 PM"
}
```

5. Created `/api/hackathon-settings` endpoint for querying and modifying the document. 

image

## Future Work (not in this PR) 

Future TODO: instead of relying on hardcoded start/end dates and times, refactor existing components to use this setting.

### CODE REVIEW

File: pages/api/hackathon-settings.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+				// find the hackathon settings
+				const updatedHackathonSettings = await hackathon.findOneAndUpdate(
+					{},
+					{
+						HACKATHON_START,
+						HACKATHON_END,
+						JUDGING_START,
+						JUDGING_END,
+					},
```
[the remark this answers is not in the document]
Comment (jacoblurie29) [reply pages/api/hackathon-settings.ts:45]: Does this matter?

**Review by jacoblurie29 — DISMISSED** [review]
Love this, my only comment would be that it is a bit... ugly 😂 Can we integrate the new theming and some friendly ant-design components as the pickers and button?

**Comment by zineanteoh** [comment]
@jacoblurie29 ok sir, does it look good now? 
image

**Review by JiashuHarryHuang — APPROVED** [review]
LGTM

**Comment by zineanteoh** [comment]
@jacoblurie29 dismissing your review, thanks for the UI improvement suggestion :)

**Pull request merged**